compiler/runtime/palloc.c

```diff
@@ -438,9 +438,21 @@ void sk_commit(char* new_root, uint32_t sync) {
 
 typedef struct {
   int64_t version;
-  void* bottom_addr;
+  struct file_mapping* bottom_addr;
```

Comment:
Would `file_mapping_t` instead of `struct file_mapping` work here?

Reply:
I thought you couldn't refer to a type defined later, I'll double-check, it may have changed since I learned C

Reply:
Isn't file_mapping also only defined later? I'm not positive, C has funky distinctions between just referring to names that might later get defined vs actually declaring names.

Reply:
Indeed, I could move the typedef earlier and keep the declaration below.